src/adapters/sqlite/makeDispatcher/index.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+/* eslint-disable global-require */
+
+import { fromPairs } from 'rambdax'
+
+import DatabaseBridge from './node/DatabaseBridge'
+
+import { type ConnectionTag, logger, invariant } from '../../../utils/common'
+
+import { fromPromise } from '../../../utils/fp/Result'
+
+import type { DispatcherType, SQLiteAdapterOptions, NativeDispatcher } from '../type'
+
+import { syncReturnToResult } from '../common'
+
+export { DatabaseBridge }
+
+const dispatcherMethods = [
+  'initialize',
+  'setUpWithSchema',
+  'setUpWithMigrations',
+  'find',
+  'query',
+  'count',
+  'batch',
+  'batchJSON',
+  'getDeletedRecords',
+  'destroyDeletedRecords',
+  'unsafeResetDatabase',
+  'getLocal',
+  'setLocal',
+  'removeLocal',
+]
+
+export const makeDispatcher = (
+  type: DispatcherType,
+  tag: ConnectionTag,
+  _dbName: string,
+): NativeDispatcher => {
+  // const jsiDb = false // type === 'jsi' && global.nativeWatermelonCreateAdapter(dbName)
+
+  const methods = dispatcherMethods.map(methodName => {
+    // batchJSON is missing on Android
+    // if (!DatabaseBridge[methodName] || (methodName === 'batchJSON' && jsiDb)) {
+    //   return [methodName, undefined]
+    // }
+
+    const name = type === 'synchronous' ? `${methodName}Synchronous` : methodName
+
+    return [
+      methodName,
+      (...args) => {
+        // $FlowFixMe
+        const callback: any => void = args[args.length - 1]
+        const otherArgs = args.slice(0, -1)
+
+        // if (jsiDb) {
+        //   try {
+        //     const value =
+        //       methodName === 'query' || methodName === 'count'
+        //         ? jsiDb[methodName](...otherArgs, []) // FIXME: temp workaround
+        //         : jsiDb[methodName](...otherArgs)
+        //     callback({ value })
+        //   } catch (error) {
+        //     callback({ error })
+        //   }
+        //   return
+        // }
+
+        // $FlowFixMe
+        // const returnValue = DatabaseBridge[name](_tag, ...otherArgs)
+
+        // if (type === 'synchronous') {
+        //   callback(syncReturnToResult(returnValue))
+        // } else {
+        //   fromPromise(returnValue, callback)
+        // }
+        if (type === 'synchronous') {
+          callback(syncReturnToResult(DatabaseBridge[name](tag, ...otherArgs)))
+        } else {
+          const promise = new Promise((resolve, reject) => {
+            DatabaseBridge[name](tag, ...otherArgs, resolve, (code, message, error) => {
+              reject(error)
+            })
+          })
+          fromPromise(promise, callback)
+        }
+      },
+    ]
+  })
+
+  const dispatcher = fromPairs(methods)
+  // $FlowFixMe
+  return dispatcher
+}
+
+const initializeJSI = () => {
+  if (global.nativeWatermelonCreateAdapter) {
```

Comment:
there's a bunch of junk like this function that really makes no sense in the context of a node implementation.

Reply:
cleaned it up